### PR TITLE
Adding some GNU options of grep

### LINKED ISSUE #1: Add options of GNU grep
It would be nice if the grep options were supported, with the same meanings. At least: -E, -F, -i, -w.

### PR DESCRIPTION
**Pull request by christopheNan**
Closes #1

### CODE REVIEW

**Comment by JulienPalard** [comment]
By default, now, given a single parameter, pogrep does nothing.

It's probably closer than what's grep's doing (grep's reading on stdin, pogrep exits silently):

```bash
$ pogrep Ł

$
```

I don't feel it's good to have to list all the files needed, for one not having globstar, it means using `pogrep Ł *.po */*.po`.

Those two remarks lead me to think: Why not behave like `git grep` more than `grep` by recusrsing by default? I mean making `pogrep Ł` behave like `grep Ł **/*.po`, so more like `git grep Ł`.

**Comment by christopheNan** [comment]
git has the default "knowledge" of the repository in which it operates. Do we assume that pogrep is working on a subtree (which might be different from the git repository) ?

**Comment by JulienPalard** [comment]
> git has the default "knowledge" of the repository in which it operates. Do we assume that pogrep is working on a subtree (which might be different from the git repository) ?

Maybe not. Maybe just give a usage instead of nothing, on missing arguments?

File: pogrep.py [hunk 1]
```diff
@@ -103,18 +104,23 @@ def parse_args():
...
+                        help="Suppress error messages about nonexistent or unreadable files.")
     parser.add_argument("pattern")
     parser.add_argument("path", action="append", type=argparse.FileType('r'), nargs='+')
     return parser.parse_args()
 
 
 def main():
     args = parse_args()
+    if args.fixed_strings:
+        for car in "\.^$*+?{}[]|()":
```
Comment (JulienPalard) [inline pogrep.py]: What about using [re.escape](https://docs.python.org/3/library/re.html#re.escape)?

Comment (christopheNan) [reply pogrep.py]: If only I RTFM...

**Comment by JulienPalard** [comment]
@christopheNan Would you mind also implementing `-r` ? I don't fell typing `pogrep supports *.po */*.po` instead of the old simple `pogrep supports`.
`pogrep -r supports` would be OK though, and like modern grep, not giving `.` if `-r` is given should be allowed.

**Comment by christopheNan** [comment]
Program would need some tests now.

File: pogrep.py [hunk 2]
```diff
@@ -1,75 +1,164 @@
...
+            sys.exit(1)
+    if exclude_dir:
+        files = [elt for elt in files if exclude_dir.rstrip(os.sep) + os.sep not in elt]
+    return files
 
 
 def parse_args():
     parser = argparse.ArgumentParser(description="Find translated words.")
+    parser.add_argument("-F", "--fixed-strings", action="store_true",
+                        help="Interpret pattern as fixed string, not regular expressions")
```
Comment (JulienPalard) [inline pogrep.py]:
```suggestion
                        help="Interpret pattern as fixed string, not regular expressions.")
```

File: pogrep.py [hunk 3]
```diff
@@ -1,75 +1,164 @@
...
+    parser.add_argument('-n', "--line-number", action="store_true",
+                        help="Prefix each line of output with the 1-based line number within its input file.")
+    parser.add_argument("-l", "--files-with-matches", action="store_true",
+                        help="Suppress normal output; instead print the name of each input file from which output "
+                             "would normally have been printed.  The scanning will stop on the first match.")
+    parser.add_argument("-s", "--no-messages", action="store_true",
+                        help="Suppress error messages about nonexistent or unreadable files.")
+    parser.add_argument("-r", "--recursive", action="store_true",
+                        help="Read  all  files under each directory, recursively, following symbolic links only "
+                             "if they are on the command line.  Note that if no file operand is given, pogrep searches"
```
Comment (JulienPalard) [inline pogrep.py]:
```suggestion
                             "if they are on the command line.  Note that if no file operand is given, pogrep searches "
```

Else we're having `searchesthe working directory`.

File: pogrep.py [hunk 4]
```diff
@@ -1,75 +1,164 @@
...
+                        help="Select only those lines containing matches that form whole words.")
+    parser.add_argument('-n', "--line-number", action="store_true",
+                        help="Prefix each line of output with the 1-based line number within its input file.")
+    parser.add_argument("-l", "--files-with-matches", action="store_true",
+                        help="Suppress normal output; instead print the name of each input file from which output "
+                             "would normally have been printed.  The scanning will stop on the first match.")
+    parser.add_argument("-s", "--no-messages", action="store_true",
+                        help="Suppress error messages about nonexistent or unreadable files.")
+    parser.add_argument("-r", "--recursive", action="store_true",
+                        help="Read  all  files under each directory, recursively, following symbolic links only "
```
Comment (JulienPalard) [inline pogrep.py]: You doubled the spaces between the three first words, interestingly it's rendered deduplicated.

Comment (christopheNan) [reply pogrep.py]: Well, I copy/pasted from ``man grep`` (I confirm there are doubled spaces on this specific line).

File: pogrep.py [hunk 5]
```diff
@@ -1,75 +1,164 @@
...
```
Comment (christopheNan) [inline pogrep.py]:
```suggestion
                        help="Read all files under each directory, recursively, following symbolic links only "
```

**Comment by JulienPalard** [comment]
Thanks a lot @christopheNan!

**Pull request merged**